src/utils/url.ts

```diff
@@ -28,8 +29,12 @@ export function isURLInList(url: string, list: string[]) {
  * @param urlTemplate URL template ("google.*", "youtube.com" etc).
  */
 export function isURLMatched(url: string, urlTemplate: string): boolean {
-    const regex = createUrlRegex(urlTemplate);
-    return Boolean(url.match(regex));
+    if (!(url.includes('[')) || !(urlTemplate.includes('['))) {
```

Comment:
Let's save the result of `includes` into some variables. If one includes and other not, return false. If both include, check IPv6, otherwise check as usual.

Reply:
Isn't that what || is for